recipes/configs/code_llama2/7B_qlora_single_device.yaml

```diff
@@ -51,36 +45,40 @@ checkpointer:
 resume_from_checkpoint: False
 save_adapter_weights_only: False
 
-# Fine-tuning arguments and training
-batch_size: 2
+# Dataset
+dataset:
+  _component_: torchtune.datasets.alpaca_cleaned_dataset
+seed: null
+shuffle: True
+
+# Fine-tuning arguments
 epochs: 1
 max_steps_per_epoch: null
-gradient_accumulation_steps: 64
-compile: False
-
+batch_size: 2
+gradient_accumulation_steps: 16
 optimizer:
   _component_: torch.optim.AdamW
   weight_decay: 0.01
   lr: 3e-4
-
 lr_scheduler:
   _component_: torchtune.modules.get_cosine_schedule_with_warmup
   num_warmup_steps: 100
-
 loss:
   _component_: torch.nn.CrossEntropyLoss
+compile: False
 
-
-# Training environment
+# Training env
 device: cuda
+
+# Memory management
 enable_activation_checkpointing: True
 dtype: bf16
 
 # Logging
+output_dir: /tmp/codellama_qlora_finetune_output
```

Comment:
every cfg is like this, but it doesnt make sense to me that:

metric_logger.log_dir: /tmp/CodeLlama-7b-hf/logs
profiler.output_dir: /tmp/CodeLlama-7b-hf/profiling_outputs

But the output_dir is not /tmp/CodeLlama-7b-hf/output

Its probably a conversation for another PR, but i thought it was worth the comment. Ideally we should have something like:

root_dir:/tmp/CodeLlama-7b-hf/

And every other path start with f"{root_dir}/path"

Reply:
Completely agree, but I think this should be handled in another PR